libs/pika/execution/include/pika/execution/algorithms/drop_operation_state.hpp

```diff
@@ -248,5 +249,16 @@ namespace pika::execution::experimental {
         {
             return detail::partial_algorithm<drop_operation_state_t>{};
         }
-    } drop_operation_state{};
+    };
+
+    /// \brief Releases the operation state of the adaptor before signaling a connected receiver.
+    /// The operation_state of previous senders can potentially hold on to a allocated memory longer
+    /// than necessary which would prevent other algorithm from accessing this resource.
```

Comment:
```suggestion
    /// The operation state of previous senders can hold on to allocated memory or values longer
    /// than necessary which can prevent other algorithms from using those resources.
```
?

Reply:
And to make it explicit, perhaps add this afterwards:
```suggestion
    /// drop_operation_state (not sure about the markup here to make it formatted correctly) can be used to explicitly release the operation state, and thus associated resources, of previous senders.
```
?